### DISCUSSION TITLE
What is `L` in the custom objectives?

### DISCUSSION THREAD

**Discussion by unary-code** [MilesCranmer/PySR#372] [Q&A] [answered]
Where does the "L" in L(Inf) come from?
How can I return a "L(Inf)" value in my Julia code in my julia helper object jl?
Currently, even when I do "using SymbolicRegression" in my Julia helper object, I can not return "L(Inf)" in my custom function "eval_loss", because when I do return "L(Inf)" inside "eval_loss" and I run the julia code of the custom function "eval_loss", the Julia interpreter gives me a error in the terminal.

For your reference, "jl" is my julia helper object, and "jl" is defined as 
```
jl = pysr.julia_helpers.init_julia(julia_kwargs={"threads": 8, "optimize": 3})
```

Specifically, when I run:
```
jl.eval("""
using SymbolicRegression
function my_func(some_x)

    return L(Inf)
end

println(my_func(3))

""")

``` 
in my Python3 terminal, the Julia interpreter gives me this error in the terminal:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/anaconda3/envs/pysr_env/lib/python3.10/site-packages/julia/core.py", line 627, in eval
    ans = self._call(src)
  File "/opt/anaconda3/envs/pysr_env/lib/python3.10/site-packages/julia/core.py", line 555, in _call
    self.check_exception(src)
  File "/opt/anaconda3/envs/pysr_env/lib/python3.10/site-packages/julia/core.py", line 609, in check_exception
    raise JuliaError(u'Exception \'{}\' occurred while calling julia code:\n{}'
julia.core.JuliaError: Exception 'UndefVarError: `L` not defined' occurred while calling julia code:

using SymbolicRegression
function my_func(some_x)

    return L(Inf)
end

println(my_func(3))


```

Why this is somewhat important: When I pass into my Julia objective function a lot of the math functions that my model.equations_ contains, the Julia helper's objective function returns a different loss value from what the model.equations_ reports as the loss value in the "loss" column for that math function.
So, I want to make my Julia helper's objective function (that I use to double-check the model's reported value for loss for a particular math function) *as close as possible* to the Julia objective function that my model "model" uses.
To make my Julia helper's objective function *as close as possible* to my model's objective function, I want to be able (in my Julia helper's objective function) to return L(Inf) instead of just Inf.

**Comment by MilesCranmer** [comment, marked as answer]
`L` is just a type variable that you have in the loss functions — it’s based on the dataset type of `Dataset{T,L}` that you would need to declare in the function signature. `L` could be a `Float32` or a `Float64` or something else. 

Note that `L(Inf)` just means: “convert infinity to type `L`”.

If you want to use `L` from the loss function in other functions, you could pass it. But normally all you need to do is just wrap the output of the other function with `L(…)` to convert whatever numerical type it outputs.

**Reply by MilesCranmer** [reply]
If you are using `precision=32`, you could also just assume `L=Float32` and use that instead
